VERSION-RANGE-SPEC.rst

```diff
@@ -0,0 +1,989 @@
+  quoting rules. This should be rare in practice.
+
+The list of ``<version-constraint>s`` of a range are signposts in the version
+timeline of a package. With these few and simple validation rules, we can avoid
+the creation of most empty or impossible version ranges:
+
+- **Constraints are sorted by version**. The canonical ordering is the versions
+  order. The ordering of ``<version-constraint>`` is not significant otherwise
+  but this sort order is needed when check if a version is contained in a range.
+
+- **Versions are unique**. Each ``version`` must be unique in a range and can
+  occur only once in any ``<version-constraint>`` of a range specifier,
+  irrespective of its comparators. Tools must report an error for duplicated
+  versions.
+
+- **There is only one star**: "*" must only occur once and alone in a range,
+  without any other constraint or version.
+
+Starting from a de-duplicated and sorted list of constraints, these extra rules
+apply to the comparators of any two contiguous constraints to be valid:
+
+- "!=" constraint can be followed by a constraint using any comparator, i.e.,
+  any of "=", "!=", ">", ">=", "<", "<=" as comparator (or no constraint).
+
+Ignoring all constraints with "!=" comparators:
+
+- A "=" constraint must be followed only by a constraint with one of "=", ">",
+  ">=" as comparator (or no constraint).
+
+And ignoring all constraints with "=" or "!=" comparators, the sequence of
+constraint comparators must be an alternation of greater and lesser comparators:
+
+- "<" and "<=" must be followed by one of ">", ">=" (or no constraint).
+- ">" and ">=" must be followed by one of "<", "<=" (or no constraint).
+
+Tools must report an error for such invalid ranges.
+
+
+Parsing and validating version range specifiers
+-------------------------------------------------
+
+To parse a version range specifier string:
+
+- Remove all spaces and tabs.
+- Start from left, and split once on colon ":".
+- The left hand side is the URI-scheme that must be lowercase.
+  - Tools must validate that the URI-scheme value is ``vers``.
+- The right hand side is the specifier.
+
+- Split the specifier from left once on a slash "/".
+
+- The left hand side is the <versioning-scheme> that must be lowercase.
+  Tools should validate that the <versioning-scheme> is a known scheme.
+
+- The right hand side is a list of one or more constraints.
+  Tools must validate that this constraints string is not empty ignoring spaces.
+
+- If the constraints string is equal to "*", the ``<version-constraint>`` is "*".
+  Parsing is done and no further processing is needed for this ``vers``. A tool
+  should report an error if there are extra characters beyond "*". 
+
+- Strip leading and trailing pipes "|" from the constraints string.
+- Split the constraints on pipe "|". The result is a list of ``<version-constraint>``.
+  Consecutive pipes must be treated as one and leading and trailing pipes ignored.
```

Comment:
I don't mind that rule for implementers. BUT: Consecutive pipes SHALL not exist in the first place.